### ISSUE TITLE
Percona INNODB key rotation fails with "permission denied"

### ISSUE THREAD

**Issue by kashak88** [hashicorp/vault#17044]
Hello,
I apologise if I've missed something in the release notes. We've used this article to set it up https://www.percona.com/blog/2018/09/17/using-the-keyring_vault-plugin-with-percona-server-for-mysql-5-7 . I remember the master key rotation worked without issues previously. We haven't done rotation in a while so I cannot pinpoint on when exactly it stopped working.
Ubuntu 20.04
Vault 1.11.2
Consul backend 1.13.1

```
mysql> ALTER INSTANCE ROTATE INNODB MASTER KEY;
ERROR 3185 (HY000): Can't find master key from keyring, please check in the server log if a keyring plugin is loaded and initialized successfully.
```
mysql-error.log:
```
2022-09-07T14:19:01.385858Z 473694 [ERROR] Plugin keyring_vault reported: 'Could not write key to Vault. Vault has returned the following error(s): permission denied'
2022-09-07T14:19:01.385902Z 473694 [ERROR] Plugin keyring_vault reported: 'Could not flush keys to keyring'
```
vault log:
```
{"time":"2022-09-07T14:31:34.9163081Z","type":"request","auth":{"token_type":"default"},"request":{"id":"a4c12ea6-12b4-d1c3-cd2b-f471ff022937","operation":"update","mount_type":"kv","client_token":"hmac-sha256:4dc72f52d8399434821626938638854a422cff0dfb50d95bebe0ab9bc5012349","namespace":{"id":"root"},"path":"secret/encryption/p2m-dev-dbs/p2m-dev-db-replica/NDhfSU5OT0RCS2V5LTI4YzY5ZjM4LTI1NTItMTFlYi1hZmQ1LTAwNTA1Njg1YTA3MS0zMF8=","data":{"type":"hmac-sha256:e8593f83150067c9386f7b757407e3709a898c09fd3d61a97ec86b4d061732c4","value":"hmac-sha256:d4d613aa547273c2b2b66775e1b85d98b5ef7ca46aecb1194a2c81d55f40ed6e"},"remote_address":"10.13.1.142","remote_port":53788},"error":"permission denied"}
{"time":"2022-09-07T14:31:34.91675373Z","type":"response","auth":{"token_type":"default"},"request":{"id":"a4c12ea6-12b4-d1c3-cd2b-f471ff022937","operation":"update","mount_type":"kv","client_token":"hmac-sha256:4dc72f52d8399434821626938638854a422cff0dfb50d95bebe0ab9bc5012349","namespace":{"id":"root"},"path":"secret/encryption/p2m-dev-dbs/p2m-dev-db-replica/NDhfSU5OT0RCS2V5LTI4YzY5ZjM4LTI1NTItMTFlYi1hZmQ1LTAwNTA1Njg1YTA3MS0zMF8=","data":{"type":"hmac-sha256:e8593f83150067c9386f7b757407e3709a898c09fd3d61a97ec86b4d061732c4","value":"hmac-sha256:d4d613aa547273c2b2b66775e1b85d98b5ef7ca46aecb1194a2c81d55f40ed6e"},"remote_address":"10.13.1.142","remote_port":53788},"response":{"mount_type":"kv","data":{"error":"hmac-sha256:519cb0764fc5235a13ecca6b83bea5dc331f049fd01fa7140e3ac045f52f2dc1"}},"error":"1 error occurred:\n\t* permission denied\n\n"}
```
vault policy:
```
vault policy read p2m_dev_tablespace_encryption
path "secret/encryption/p2m-dev-dbs/*" {
  capabilities = ["create", "read", "delete", "update", "list"]
}
```
vault cert auth that percona server uses:
```
Key                             Value
---                             -----
allowed_common_names            []
allowed_dns_sans                []
allowed_email_sans              []
allowed_metadata_extensions     <nil>
allowed_names                   []
allowed_organizational_units    []
allowed_uri_sans                []
bound_cidrs                     [10.13.1.0/24]
certificate                     -----BEGIN CERTIFICATE-----
--REDACTED--
-----END CERTIFICATE-----
display_name                    p2m_dev_encryption
policies                        [p2m_dev_tablespace_encryption]
required_extensions             []
token_bound_cidrs               [10.13.1.0/24]
token_explicit_max_ttl          0s
token_max_ttl                   0s
token_no_default_policy         false
token_num_uses                  0
token_period                    0s
token_policies                  [p2m_dev_tablespace_encryption]
token_ttl                       168h3m20s
token_type                      default
ttl                             168h3m20s
```
keyring_vault.conf:
```
cat keyring_vault.conf 
vault_url = https://vault1.util.redacted.com:8200
secret_mount_point = secret/encryption/p2m-dev-dbs/p2m-dev-db-replica
token = REDACTED
```
All the other functionalities work without issues with the above configuration. Please let me know if there is anything else that I might be missing

**Comment by kashak88** [comment]
hello, any update on this?

**Comment by kashak88** [comment]
Looks like this has been fixed with 1.12.0...closing
